### PR TITLE
Allow hotlinks to steps - the proper way (IMO)

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
please review and test this - we need to get this sorted as the unfinished work was deployed on o3 ;(

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 1]
```diff
@@ -103,6 +103,13 @@ sub needle_url {
 # Call to viewimg or viewaudio
 sub view {
     my ($self) = @_;
+
+    # Redirect users with the old preview link
+    if (!$self->req->is_xhr) {
+        my $target_url = $self->url_for('test', 'testid' => $self->param('testid'))."?".$self->param('moduleid')."/steps/".$self->param('stepid');
+        return $self->redirect_to($target_url);
+    }
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Step.pm]: where is the test for this? you rewrote the test to rely on the new way only, but can you preserve the deprecated way in tests, too?

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 2]
```diff
@@ -103,6 +91,14 @@ sub needle_url {
 # Call to viewimg or viewaudio
 sub view {
     my ($self) = @_;
+
+    # Redirect users with the old preview link
+    if (!$self->req->is_xhr) {
+        my $anchor = "#step/" . $self->param('moduleid') . "/" . $self->param('stepid');
+        my $target_url = $self->url_for('test', testid => $self->param('testid'));
+        return $self->redirect_to($target_url . $anchor);
+    }
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Step.pm]: where is the test for this? you rewrote the test to rely on the new way only, but can you preserve the deprecated way in tests, too?

**Comment by okurz** [comment]
Regarding functionality: I am testing this locally

This is a huge bunch of uncommented and hacky javascript. I recommend to add more tests for this, considering that this is basically a fix for the previous PR which broke a lot. So tests are missing, obviously. Can you please add some tests?

**Comment by coolo** [comment]
we're at 66% coverage for a reason - unfortunately ;(

Tests for interaction with javascript are still very tricky, but having a test for the backward compat (thinking old bugzilla links) are part of the TODO for #726, so I guess I have to do it here too ;)

**Comment by okurz** [comment]
see my comment above? It has a proposal for a test and it works

**Comment by coolo** [comment]
yes, I did the same - without highjacking unrelated tests ;)

9f6240e

**Comment by okurz** [comment]
just moving test calls into another file does not make a difference in relation to existing tests, but it does slow down testing.

**Comment by okurz** [comment]
why not use "subtests"? You could at least merge the two new files

**Comment by coolo** [comment]
I don't like subtests - they clutter the backtrace

**Comment by coolo** [comment]
anyway, +0.3%

**Pull request merged**